art/estimators/generation/tensorflow1.py

```diff
@@ -0,0 +1,167 @@
+               the shape of clip values needs to match the total number of features.
+        :type clip_values: `tuple`
+        :param preprocessing_defences: Preprocessing defence(s) to be applied by the classifier.
+        :type preprocessing_defences: :class:`.Preprocessor` or `list(Preprocessor)` instances
+        :param postprocessing_defences: Postprocessing defence(s) to be applied by the classifier.
+        :type postprocessing_defences: :class:`.Postprocessor` or `list(Postprocessor)` instances
+        :param preprocessing: Tuple of the form `(subtractor, divider)` of floats or `np.ndarray` of values to be
+               used for data preprocessing. The first value will be subtracted from the input. The input will then
+               be divided by the second one.
+        :type preprocessing: `tuple`
+        :param feed_dict: A feed dictionary for the session run evaluating the classifier. This dictionary includes all
+                          additionally required placeholders except the placeholders defined in this class.
+        :type feed_dict: `dictionary`
+        """
+
+        super(Tensorflow1Gan, self).__init__(
+            clip_values=clip_values,
+            channel_index=channel_index,
+            preprocessing_defences=preprocessing_defences,
+            postprocessing_defences=postprocessing_defences,
+            preprocessing=preprocessing,
+        )
+
+        self._input_ph = input_ph
+        self._encoding_length = self._input_ph.shape[1]
+        self._model = model
+        self._loss = loss
+        if self._loss is not None:
+            self._grad = tf.gradients(self._loss, self._input_ph)
+        self._feed_dict = feed_dict
+
+        # Assign session
+        if sess is None:
+            raise ValueError("A session cannot be None.")
+            # TODO do the same thing for all not None variables
+        self._sess = sess
+
+    def loss(self, z, image_adv):
+        pass
+
+    def predict(self, x):
+        """
+        Perform projections over a batch of encodings.
+
+        :param x: Encodings.
+        :type x: `np.ndarray`
+        :return: Array of prediction projections of shape `(num_inputs, nb_classes)`.
+        :rtype: `np.ndarray`
+        """
+        logging.info("Projecting new sample from z value")
+        y = self._sess.run(self._model, feed_dict={self._input_ph: x})
+        return y
+
+    def loss_gradient(self):
+        pass
+
+    def fit(self, x, y, batch_size=128, nb_epochs=10, **kwargs):
+        """
+        do nothing.
+        """
+        pass
+
+    def get_activations(self, x, layer, batch_size=128):
+        """
+        do nothing.
+        """
+        pass
+
+    def set_learning_phase(self, train):
+        """
+        do nothing.
+        """
+        pass
+
+    @property
+    def generator_model(self):
+        """
+        Returns the generator tensor
+        :return:
+        :rtype: 'float'
+        """
+        return self._model
```

Comment:
Would it be possible to use the existing Estimator property `model` instead of adding `generator_model`?

Reply:
yes, that's what I had originally called it but for some reason it was causing a bug. I don't remember what it was but I just tried renaming it now and it seems to work so who knows what was going on there :)